### ISSUE TITLE
createElement does not warn about props.__proto__ being null, fails instead

### ISSUE THREAD

**Issue by alitaheri** [facebook/react#6854]
This is a bug in `React.createElement` code:

`React.createElement` does not warn if `props.__proto__` is null ( from `Object.create(null)` ) since it [believes that to be a plain object](https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L135) originating from #6134. But since it directly accesses `hasOwnProperty` it fails if such object is passed as props.

Reproduce:

```js
const props = Object.create(null);
React.createElement(() => null, props);
```

**What is the expected behavior?**
It shouldn't use `hasOwnProperty` directly on props if such object is expected.

React: 15.1.0
Browsers: All
OS: All

This affects all versions of React but since after 15.1.0 the acceptable shape of props is explicitly defined then it should be explicitly supported as well.

**Comment by bloodyowl** [comment]
how about just use `Object.prototype.hasOwnProperty.call(config, key)`?

**Comment by gaearon** [comment]
>how about just use Object.prototype.hasOwnProperty.call(config, key)?

Sounds good, would you like to send a PR with a fix and a test case?

**Comment by alitaheri** [comment]
@gaearon Sure, but i need to know how would the community want it fix, support prototype-less objects or just fail with reason.

**Comment by gaearon** [comment]
>+1 for support prototype-less objects

Sure, I don’t see why not. I think we did support them, and we just regressed when we added a warning. Thanks!

**Comment by gaearon** [comment]
Fixed in #6855.
